src/NetMQ/zmq/Transports/Tcp/TcpConnector.cs

```diff
@@ -180,20 +189,22 @@ private void StartConnecting()
         }
 
         /// <summary>
-        /// 
+        /// This method is called when a message Send operation has been completed.
         /// </summary>
-        /// <param name="socketError"></param>
-        /// <param name="bytesTransferred"></param>
-        /// <exception cref="NetMQException">a non-recoverable socket error occurred.</exception>
+        /// <param name="socketError">a SocketError value that indicates whether Success or an error occurred</param>
+        /// <param name="bytesTransferred">the number of bytes that were transferred</param>
+        /// <exception cref="NetMQException">A non-recoverable socket error occurred.</exception>
         public void OutCompleted(SocketError socketError, int bytesTransferred)
         {
+            m_ioObject.RemoveSocket(m_s);
+            m_isHandleValid = false;
+
             if (socketError != SocketError.Success)
             {
-                m_ioObject.RemoveSocket(m_s);
```

Comment:
I don't remember doing that. Did I err in trying to merge from the master, before committing? If not - then this is an error. I had no intention of changing that.